### ISSUE TITLE
Chunks don't load

### ISSUE THREAD

**Issue by Kuzmaniac** [ItemsAdder/ItemsAdderBlocksInjector#8]
Hey!
whenever i add a custom block in the json files of iris the world doesn't load. sometimes it works but then only the chunk that contains the nine blocks doesn't load. with the standard iris files it works great but not with the new ones from ItemsAdder

Server on Paper 1.19.2
ItemsAdder_3.3.0b-r5
itemsadder-blocks-injector 0.0.7
Iris-2.3.13-1.19.3

**Comment by bowencraft** [comment]
I have the same issue, as after I add custom block in the iris pack, the new chunk won't load. I haven't try if only chunk with custom blocks not loaded or all the chunks are not loading anymore. There is no error message in the console.
<img width="1572" alt="IMG10644" src="https://user-images.githubusercontent.com/34801660/215544083-c543f472-2920-46a7-86e0-d3c72730d76d.png">
For more information, I tested on server Paper 1.19.2, I used ItemsAdder_3.3.1, itemsadder-blocks-injector 0.0.7, and I tried both Iris-2.3.13-1.19.3 and Iris-2.3.9-1.19.2.

**Comment by freezyXD** [comment]
same problem

**Comment by Kuzmaniac** [comment]
:/

**Comment by thefourcraft** [comment]
I am going to test it as soon as I have the time, I am using iris on production and I don't have any issues

my server is on 1.19.2 paper and I am using iris 2.3.13-1.19.3-hotfix 
going to test as soon as I have the time

**Comment by Kuzmaniac** [comment]
I hope the bug will be fixed!

**Comment by thefourcraft** [comment]
This is an issue that was interesting while testing... it was very random
if you use `/pregen` on IRIS, you don't get this issue, but if you don't, you will have this issue but not all the time
I tested it on 1.19.2 using a paper ([#386](https://api.papermc.io/v2/projects/paper/versions/1.19.3/builds/386/downloads/paper-1.19.3-386.jar))
Used ItemsAdder 3.3.1, itemsadder-blocks-injector 0.0.7

There is an issue, just not sure where... also not a related bug. Some mm mobs are spawning wrong 😕 

I need to do more testing to figure out the issue, maybe turning on debug on both plugins whloud yeld something 🤞🏻

**Comment by Kuzmaniac** [comment]
I just tried it again but it doesn't work with debug either.

![2023-02-01_17 39 32](https://user-images.githubusercontent.com/107998138/216105905-306a3ac3-00fe-47a7-9096-0598afa53f3d.png)
Paper 1.19.2
ItemsAdder 3.3.1
ItemsAdder-Blocks-Injector 0.0.7
Iris 2.3.13-1.19.3-hotfix

**Comment by thefourcraft** [comment]
> I just tried it again but it doesn't work with debug either.

debug gives more info about the plugin and what's happening it doesn't solve the bug, it helps identify the issue

**Comment by bowencraft** [comment]
I tried with `iris pregenerate` but it seems also not working.
To be more specific, the itemsadder custom block can be normally identified and autocompleted in the iris studio IDE, however the block cannot be summon.

Here is the code snippet I used in the iris pack:
```json
{
    "name": "Empty",
    "derivative": "THE_VOID",
    "vanillaDerivative": "THE_VOID",
    "layers": [
        {
            "palette": [
                {
                    "block": "iasurvival:modern_stone",
                    "weight": 200
                },
                {
                    "block": "minecraft:white_concrete",
                    "weight": 50
                }
            ]
        }
    ],
    "generators": [
        {
            "generator": "empty",
            "min": -40,
            "max": 60
        }
    ]
}
```

**Comment by thefourcraft** [comment]
@NextdoorPsycho you got any ideas why this whloud happen?

**Comment by T1Doggo** [comment]
I do also have the issue, hope it gets fixed soon :(
Using Paper 1.19.2
ItemsAdder 3.3.1
ItemsAdder-Blocks-Injector 0.0.7
Iris 2.3.13-1.19.3-hotfix

**Comment by LoneDev6** [comment]
Will be tested again, probably compatibility broke with some latest Iris update.
Please provide a valid Iris configuration folder for me to load it and reproduce the issue.

**Comment by LoneDev6** [comment]
I also need a full server log.

**Comment by T1Doggo** [comment]
> Will be tested again, probably compatibility broke with some latest Iris update.
> Please provide a valid Iris configuration folder for me to load it and reproduce the issue.

Where do I send you this?

**Comment by CocoTheOwner** [comment]
[> ](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/issues/8#issuecomment-1411595451)

Can you elaborate on this? Do you mean that if you use _another_ pregeneration tool such as Chunky it happens, or if you are generating chunks live (i.e. no pregeneration happens at all)?
@thefourcraft

**Comment by LoneDev6** [comment]
@T1Doggo 
> Where do I send you this? The config was fresh installed 

Basically you're telling me that the default Iris configurations (with no custom block from itemsadder added) is not loading correctly when itemsadder-blocks-injector is installed?

**Comment by T1Doggo** [comment]
> @T1Doggo
> 
> > Where do I send you this? The config was fresh installed
> 
> Basically you're telling me that the default Iris configurations (with no custom block from itemsadder added) is not loading correctly when itemsadder-blocks-injector is installed?

Oh no I meant that the default config with a custom block on the overworld.json and iablockinjector don't load so basically you can just add a custom block to the iris config and I guess you'll get the same issue

**Comment by LoneDev6** [comment]
> iablockinjector don't load

What do you mean? @T1Doggo

**Comment by thefourcraft** [comment]
@CocoTheOwner I can't reproduce the issue perfectly, each time, it's random.

also, @RePixelatedMC on discord told me that 

> Update to 1.19.3 1.19.2 is a unsported version and wont get updates anymore

@LoneDev6 the issue is, from what I understood, they are adding a custom block and the generation stops working

From my testing, the generation works as usual, I also always pregen the world to 100K so I don't have any loading issues

whiteout and when adding custom block's the generation is slow when using a 6GB server 

my testing server is 20GB RAM and I7 8th generation

**Comment by T1Doggo** [comment]
> > iablockinjector don't load
> 
> What do you mean? @T1Doggo

I mean the world do not load, i'm stuck in the loading screen while tping to the world. here's the full server logs tested on Purpur-1.19.2-1858, with Iris-2.3.9-1.19.2, ItemsAdder_3.3.1 and itemsadder-blocks-injector_0.0.7 
https://mclogs.minestrator.com/g7VCSt2

**Comment by thefourcraft** [comment]
@T1Doggo 
`java -Xms2G -Xmx2G -jar purpur-1.19.2-1858.jar --nogui` to use iris you need 4GB ram in order for your server to work properly 

I recommend 8 GB ram and a CPU from 2013+...

**Comment by freezyXD** [comment]
> @T1Doggo `java -Xms2G -Xmx2G -jar purpur-1.19.2-1858.jar --nogui` to use IRIS you need 4GB ram in order for your server to work properly
> 
> I recommend 8 GB ram and a CPU from 2013+...

I have same problem with 20GB RAM

**Comment by Rakise** [comment]
> @LoneDev6 
> 
> > Where do I send you this? The config was fresh installed
> 
> Basically you're telling me that the default Iris configurations (with no custom block from itemsadder added) is not loading correctly when itemsadder-blocks-injector is installed?

No. I know what is happening with him. Basically if you have a world of Iris that was generated without custom block, and puts the custom blocks later, only chunks without custom block loads. But if you create the world with custom blocks, it doesn't load into vanilla minecraft or optifine


(Fabric and forge works 100%, bug only happens on Vanilla and optifine)

**Comment by thefourcraft** [comment]
>  it doesn't load into vanilla minecraft or optifine

I'm so confused right now 
@rakise

**Comment by Rakise** [comment]
> > it doesn't load into vanilla minecraft or optifine
> 
> I'm so confused right now @Rakise
> 
> > Fabric and forge works 100%, bug only happens on Vanilla and optifine
> 
> We don't support fabric or forge with IA

Fabric, forge and optifine client***

**Comment by Rakise** [comment]
@thefourcraft
> I'm so confused right now @Rakise

Test with this:


1. Install paper or pufferfish 1.19.2 

2. Install itemsadder, Iris and itemsadderblocksinjector

3. Edit the Iris overworld config and add a custom block to the config

4. Join using Vanilla or optifine

5. Do: /Iris studio open or /Iris create

Result:
 Infinite client world generation or invisible chunks

6. Do the same but join using Fabric and sodium

Result:
 Working 100%

**Comment by Kuzmaniac** [comment]
Nice if it works with Fabric, but it would be great if it would also work with the normal Minecraft Vanilla

**Comment by Rakise** [comment]
> Nice if it works with Fabric, but it would be great if it would also work with the normal Minecraft Vanilla

It's useless work if you need to use a modded client to play in a specific server

**Comment by T1Doggo** [comment]
> @thefourcraft
> 
> > I'm so confused right now @Rakise
> 
> Test with this:
> 
> 1. Install paper or pufferfish 1.19.2
> 2. Install itemsadder, Iris and itemsadderblocksinjector
> 3. Edit the Iris overworld config and add a custom block to the config
> 4. Join using Vanilla or optifine
> 5. Do: /Iris studio open or /Iris create
> 
> Result: Infinite client world generation or invisible chunks
> 
> 6. Do the same but join using Fabric and sodium
> 
> Result: Working 100%

Already tried with fabric and sodium and the issue is still there

**Comment by Kuzmaniac** [comment]
:/

**Comment by T1Doggo** [comment]
anyway I think that the issue is from ItemsAdderBlockInjector, everything in iris is fine, same about iA

**Comment by Kuzmaniac** [comment]
Yes I believe that too

**Comment by T1Doggo** [comment]
Still getting this error when trying to pregen the world with iris, 

https://mclogs.minestrator.com/Hl1gQJg

**Comment by Rakise** [comment]
@T1Doggo

Try with 1.19 (not the .1 or .2) Fabric sodium

**Comment by NextdoorPsycho** [comment]
> I tried with `iris pregenerate` but it seems also not working.
> 
> To be more specific, the itemsadder custom block can be normally identified and autocompleted in the iris studio IDE, however the block cannot be summon.
> 
> 
> 
> Here is the code snippet I used in the iris pack:
> 
> ```json
> 
> {
> 
>     "name": "Empty",
> 
>     "derivative": "THE_VOID",
> 
>     "vanillaDerivative": "THE_VOID",
> 
>     "layers": [
> 
>         {
> 
>             "palette": [
> 
>                 {
> 
>                     "block": "iasurvival:modern_stone",
> 
>                     "weight": 200
> 
>                 },
> 
>                 {
> 
>                     "block": "minecraft:white_concrete",
> 
>                     "weight": 50
> 
>                 }
> 
>             ]
> 
>         }
> 
>     ],
> 
>     "generators": [
> 
>         {
> 
>             "generator": "empty",
> 
>             "min": -40,
> 
>             "max": 60
> 
>         }
> 
>     ]
> 
> }
> 
> ```

first of all holy shit turn down your weight values. 1 concrete to 4 modern stone is the same thing and will yield you like 3000% more performance.

**Comment by NextdoorPsycho** [comment]
and I'm not entirely sure what the problem is here. If I understand this correctly, you start your generation without custom blocks. Then you add custom blocks to your generation, and it breaks. Why not start with the custom blocks already added in the generation instead of shoehorning at it? There's a red label warning for force updating worlds for a reason or did I miss something, or do I not understand some thing about the situation? does this happen in the studio?

**Comment by NextdoorPsycho** [comment]
> > @T1Doggo `java -Xms2G -Xmx2G -jar purpur-1.19.2-1858.jar --nogui` to use IRIS you need 4GB ram in order for your server to work properly
> 
> > 
> 
> > I recommend 8 GB ram and a CPU from 2013+...
> 
> 
> 
> I have same problem with 20GB RAM

you can't just throw ram at your problems to make them go away, I don't have this problem, nor can I replicate it, which is problematic for trying to figure out how to help

**Comment by NextdoorPsycho** [comment]
> @thefourcraft
> 
> > I'm so confused right now @Rakise
> 
> 
> 
> Test with this:
> 
> 
> 
> 
> 
> 1. Install paper or pufferfish 1.19.2 
> 
> 
> 
> 2. Install itemsadder, Iris and itemsadderblocksinjector
> 
> 
> 
> 3. Edit the Iris overworld config and add a custom block to the config
> 
> 
> 
> 4. Join using Vanilla or optifine
> 
> 
> 
> 5. Do: /Iris studio open or /Iris create
> 
> 
> 
> Result:
> 
>  Infinite client world generation or invisible chunks
> 
> 
> 
> 6. Do the same but join using Fabric and sodium
> 
> 
> 
> Result:
> 
>  Working 100%

if you use purpur we hook into a natively to optimize the chunk generation, anything else, and you're rolling the dice. for example, the reason we push for it so hard is that I have literally zero issues with things like this when I'm using purpur

**Comment by NextdoorPsycho** [comment]
OK, after reading the entire thread, I am still very not understanding the problem. I understand that it has something to do with chunks not loading. Can someone please explain to me in English how they are causing and replicating the issue because I am very sick right now, and I'm just having a hard time following.

Thanks. Feel free to message me on discord

**Comment by NextdoorPsycho** [comment]
Not my problem.
Here are some notable changes:
**The placement code has been changed**
**To improve loading performance, block, and item textures are now loaded before they are processed by block and item models.
By default, textures not in the textures/item and textures/block directories will no longer be automatically recognized and will fail to load.**
**Migrated linear algebra types to Java OpenGL Math Library (JOML).** (might change lots more than we know)
**Inconsistency: Blocks are not placed correctly**
**Worldgen data packs don't work on servers at the first launch.**
LITERALLY FIXES THAT:
MC-248926 – Setting spectatorsGenerateChunks to false and logging freezes the game on the Loading Terrain screen.
MC-249059 – Loading terrain screen cannot close before 2 seconds have passed.
https://minecraft.fandom.com/wiki/Java_Edition_1.19.3

This is not our problem and is literally a game bug. Update = fix. Simple as that. They made 1.19.3 for a reason use it.
@LoneDev6  Feel free to close this once you look it over. or let me know if Im missing something

**Comment by LoneDev6** [comment]
> OK, after reading the entire thread, I am still very not understanding the problem. 

Same, and there is a lot of spam going on here.

To the reporters: Please post a proper Iris folder that I can drag and drop into my plugins folder so that I can test the issue.
Make sure to use ItemsAdder DefaultPack blocks for your Iris test.

Thanks.

**Comment by freezyXD** [comment]
@LoneDev6 , BlockInjector supports 1.19.3? I cannot start plugin on Paper 1.19.3.

**Comment by LoneDev6** [comment]
> @LoneDev6 , BlockInjector supports 1.19.3? I cannot start plugin on Paper 1.19.3.

Open a separate issue report then.
Do not report issues on others reports.

**Comment by Rakise** [comment]
@LoneDev6


I'll try to explain it as simply as possible. If you create a world using Iris with custom blocks created by itemsadder (you can open the world in the studio as well), the world will not load in vanilla. But the world exists. How it works? Vanilla stays on the "generating terrain" screen but the client works 100% It plays sounds, it works packets that the player sends to the server and also takes damage, it is 100% functional, but the screen is stuck on that message "generating terrain". How do you prove that the world exists and is functional? Using Fabric. If you log in with a Fabric client, it loads the world with custom blocks created by itemsadder (I tested it on 1.19 with a paper, purpur, pufferfish, mirai. All in 1.19.2). Now there is another problem, if the chunk that the Vanilla player spawns, there is no custom block (literally 0 custom blocks) this chunk works normally and the Vanilla player can use it (spawn in it, interact and break it). But chunks with custom blocks are invisible and don't load only in vanilla clients

**Comment by LoneDev6** [comment]
> ### To the reporters
>
>Please post a proper Iris folder that I can drag and drop into my plugins folder so that I can test the issue.
>Make sure to use ItemsAdder DefaultPack blocks for your Iris test.
>
>And please stop messaging with biased and unprecise information.
>I need proper info in order to reproduce the issue and fix it.
>
>Thanks.

**Comment by bowencraft** [comment]
> > ### To the reporters
> > Please post a proper Iris folder that I can drag and drop into my plugins folder so that I can test the issue.
> > Make sure to use ItemsAdder DefaultPack blocks for your Iris test.
> > And please stop messaging with biased and unprecise information.
> > I need proper info in order to reproduce the issue and fix it.
> > Thanks.
> 
> I am asking the same thing over and over but no one seems to understand that I need what I asked in order to identify the issue. If you keep ignoring my messages you won't get anything fixed, seems obvious to me.


Here is a simple iris package I changed from [irisdemension/emtpy](https://github.com/IrisDimensions/empty), the only change I made was to specify the layers in `biomes/empty.json` to **iasurvival:modern_stone**. 

[empty.zip](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10672334/empty.zip)

When use this pack to create a new world, on the server side, everything generates fine, no errors or warnings are reported. World generate like this way:
<img width="851" alt="screenshot_1" src="https://user-images.githubusercontent.com/34801660/217169575-c43eb2e1-386b-47a7-891c-2f59e2338189.png">

On the client side, at the forge/fabric client, you can also enter the world normally and everything works find. However, in the vanilla client, when you enter the world, your window is frozen in the loading terrains page and never finishes.
<img width="849" alt="screenshot_2" src="https://user-images.githubusercontent.com/34801660/217169824-8e1aaeae-2d38-412c-a608-6109bf3cd51b.png">

On the server side, all blocks are loaded normally. I think this may be a bug related to the client side, or as [NextdoorPsycho] said, a bug from minecraft. Thanks.

**Comment by NextdoorPsycho** [comment]
It is a Minecraft bug. And as I have stated, just update the damn game. There's nothing either of us can do if it's a game bug. The last guy that mentioned this to me I told him to update and it worked fine, just do that be the latest version of the game for  latest fixes.

**Comment by T1Doggo** [comment]
> It is a Minecraft bug. And as I have stated, just update the damn game. There's nothing either of us can do if it's a game bug. The last guy that mentioned this to me I told him to update and it worked fine, just do that be the latest version of the game for latest fixes.

I'm not against updating but how about ItemsadderBlockInjector not working on 1.19.3

**Comment by thefourcraft** [comment]
Dear reporters, @bowencraft @Rakise @T1Doggo @Kuzmaniac 

I would be grateful if you provided a complete Iris folder for testing purposes. 
This will allow me to accurately and effectively resolve any issues. 
Please use the ItemsAdder DefaultPack blocks in your testing, and provide precise and impartial information.

This thread is too long about an issue that I can't reproduce, **I have no problems on 1.19.3** **EVERYTHING WORKS FOR ME**

Regarding the Minecraft bug, updating the latest version, 1.19.3, should resolve the issue. 
Several significant changes and improvements have been made in this update, such as improved loading performance and block and item textures, migration of linear algebra types to Java OpenGL Math Library, and updates to the placement code. You can find more information about these changes at https://minecraft.fandom.com/wiki/Java_Edition_1.19.3.

> I am asking the same thing repeatedly, but no one seems to understand that I need what I asked to identify the issue.
If you ignore my messages, you won't get anything fixed, which seems obvious to me.

This is how all the messages are going... we would all like to solve your issues,  so please work together with @LoneDev6 @NextdoorPsycho

**Comment by bowencraft** [comment]
Here is a complete iris folder, used `iasurvival:modern_stone` as the custom block.
[Iris.zip](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10678226/Iris.zip)
<img width="851" alt="screenshot_1" src="https://user-images.githubusercontent.com/34801660/217332539-e7eb2f97-98b1-4e1a-ab28-f2d8b70ee0b1.png">
This is how the world should be generated like in normal situation.

I believe the main arguments about this issue now is that using 1.19.3 would solve this problem. Nevertheless, ItemsAdderBlockInjector doesn't really support 1.19.3, so we can't actually use 1.19.3. I submitted a new issue on this subject: #9 
Thanks a lot for working on it.

**Comment by Kuzmaniac** [comment]
I have the same problem with 1.19.3. The ItemsAdderBlocksInjector plugin (0.0.7) is not showing up when I do /pl, so I'm trying the 1.19.2


> Here is a complete iris folder, used `iasurvival:modern_stone` as the custom block. [Iris.zip](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10678226/Iris.zip) <img alt="screenshot_1" width="851" src="https://user-images.githubusercontent.com/34801660/217332539-e7eb2f97-98b1-4e1a-ab28-f2d8b70ee0b1.png"> This is how the world should be generated like in normal situation.
> 
> I believe the main arguments about this issue now is that using 1.19.3 would solve this problem. Nevertheless, ItemsAdderBlockInjector doesn't really support 1.19.3, so we can't actually use 1.19.3. I submitted a new issue on this subject: #9 Thanks a lot for working on it.

**Comment by Kuzmaniac** [comment]
Here the server logs and ItemAdder folder from a test server 1.19.2
The Iris Odner is too big, so once the pack folder and the rest
Logs -> [latest.log](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10684472/latest.log)
Irispack -> [packs.zip](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10684484/packs.zip)
Iris -> [Iris.zip](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10684499/Iris.zip)
ItemsAdder -> [ItemsAdder.zip](https://github.com/ItemsAdder/ItemsAdderBlocksInjector/files/10684532/ItemsAdder.zip)

**What I changed in the iris pack:**
![Bildschirmfoto 2023-02-08 um 11 24 23](https://user-images.githubusercontent.com/107998138/217507933-93dfe961-fd97-4608-9735-0f8315775baa.png)

**The result:**
![2023-02-08_11 16 43](https://user-images.githubusercontent.com/107998138/217508162-fe3cc2f7-6704-4d86-b1c2-01bb0cfa2a98.png)

**Comment by Rakise** [comment]
@thefourcraft

You guys cant fix using bukkit api or NMS. The only fix i found is changing how the original jar works when register chunks.

This issue needs to be closed. Update to 1.19.3 (client) since it's a 1.18-1.19.2 bug

**Comment by Kuzmaniac** [comment]
@Rakise 
-> https://github.com/ItemsAdder/ItemsAdderBlocksInjector/issues/9

**Comment by Rakise** [comment]
> @Rakise -> #9

> Update to 1.19.3 (client)

THIS IS A 1.18-1.19.2 CLIENT BUG

You can use the 1.19.2 server, it's a 1.19.2 client bug. Use new client versions

**Comment by LoneDev6** [comment]
Thanks for the configuration @bowencraft 

https://streamable.com/j5j5w2
Chunks load fine as you shown and as I show in the video
You can download the new blocks injector jar 0.0.8 from here which fixes 1.19.3 support.
https://github.com/ItemsAdder/ItemsAdderBlocksInjector/releases/tag/0.0.8
